_includes/header.html

```diff
@@ -0,0 +1,28 @@
+<header class="site-header">
+
+    <div class="wrapper">
+      {%- assign default_paths = site.pages | map: "path" -%}
+      {%- assign page_paths = site.header_pages | default: default_paths -%}
+      {%- assign titles_size = site.pages | map: 'title' | join: '' | size -%}
+      <a class="site-title" rel="author" href="{{ "/" | relative_url }}">{{ site.title | escape }}</a>
+  
+      {%- if titles_size > 0 -%}
```

Comment:
Did you test this tempting because I've never seen these tags written with %-? I've usually seen them written like `{% if titles_size > 0 %}`

Reply:
They're in the default template. They remove whitespace before and after the generated HTML, respectively.